### ISSUE TITLE
Timeline Markers (Today, Cursor, Custom)

### ISSUE THREAD

**Issue by andy-dorman** [namespace-ee/react-calendar-timeline#327]
I’d like to be able to drop "markers" on to the timeline which would render as a "timeline marker", pretty much like the "Today line" marker works at the moment. I'd be more than happy to handle these outside of the component itself, but if I could build an array of "markers" with corresponding time properties I could do things like set up "ctrl-[NUMBER]" key commands to jump directly to points in the timeline. I suggesting you essentially abstract out the "Today line" marker that gets set when the component initialises so you can create such markers programatically, using them pretty much like bookmarks. So for example, I've already set it up so double clicking on the canvas creates a new item at that point in time, I'd like to be able to create a new "marker" in the same way. Ideally you'd be able to bypass the snapping configuration in many of the use-cases for this stuff, but sometimes you wouldn't, so passing both the rounded values and the actual values would be good.

**Comment by mcMickJuice** [comment]
This is a good idea.  I'd like to implement this with the following goals in mind:

* user maintains state of this feature
* user is in control of rendering of this feature
* api is as declarative as possible and hopefully doesn't introduce new props on the Timeline component.

Here's the end API I have in mind:

```jsx
import Timeline, {TimelineMarker} from 'react-calendar-timeline'

<Timeline>
   // tell the timeline what and how to render TimelineMarker
   {timelineMarkers.map(marker => <TimelineMarker key={marker.id} time={marker.time}>
         // any handlers, or whatever you want in here
       <div></div>
       </TimelineMarker>
    }
</Timeline
```



The `TodayMarker` and `CursorMarker` would be handled in the exact same way:

```jsx
import Timeline, {TodayMarker, CursorMarker} from 'react-calendar-timeline'

// no need to pass in `showCursorLine` anymore
// today line won't appear by default
<Timeline>
  <TodayMarker /> 
  <CursorMarker />

  // can optionally pass in render prop/child as function to change how line is rendered
  <TodayMarker>
  // render whatever you want here
  {(style, time) => <div style={style} />}
  </TodayMarker>
</Timeline>
```

If you want more control over whats rendering, or maybe you wanna attach handlers:

```jsx
<Timeline>
  // can optionally pass in render prop/child as function to change how line is rendered
  <TodayMarker>
  // render whatever you want here...within reason though since timeline does attach certain properties to this element
  {(style, time) => <div style={style} />}
  </TodayMarker>
</Timeline>
```

To your specific use case, I think this approach would work:
* you would be in charge of making and maintaining markers, meaning that you would store this state and define when a marker is created (i.e. `onCanvasDoubleClick`)
* I'm not going to take dragSnap into account.  I believe that this is something that the user can accommodate for when determining time for marker
* the "bookmarking feature" is cool but separate from this feature (although I think you just mentioned it as it relates to your direct use case).  Sidenote: this "jump to" feature could be done using `visibleTime` props or the Timeline component instance could have a "jumpToTime" method that can be called by parent component

Thoughts on the above?

**Comment by jabidof** [comment]
I’m using another lib (google-map-react) that has this approach too. Any marker in that case needs GPS coordinates to be rendered on the map. It makes the code easier to read.

**Comment by andy-dorman** [comment]
@mcMickJuice That sounds pretty much spot on to me. The cases I gave were just for illustration, I’m sure this would give me what I need. I could probably hack region selection together with this as well while it was being properly developed.

**Comment by mcMickJuice** [comment]
Released in 0.18.0
